### PR TITLE
RT-545: scenario assessments

### PR DESCRIPTION
**Pull request by NickFitz**
Quite a few more than RT-545 in here TBH, but there's only a week left and this needs to be up ASAP :-/

### CODE REVIEW

File: update_supply_chain_information/chain_details/templates/chain_details_info.html [hunk 1]
```diff
@@ -42,6 +42,7 @@ <h3 class="govuk-heading-s" style="padding-bottom: 10px;">Description</h3>
 <h2>Space for sub components</h2>
 
 <br>
+    {% include "scenario_assessment.html" %}
```
Comment (dev-digital) [inline update_supply_chain_information/chain_details/templates/chain_details_info.html:45]: We could create a sub-folder for _includes_ now that we are going to have many files in there

File: update_supply_chain_information/chain_details/templates/gsc_notes_section.html [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{% load gsc_notes %}
+<div class="app-dit-gsc-notes">
+    {% gsc_notes instance=sc.scenario_assessment %}
```
Comment (dev-digital) [inline update_supply_chain_information/chain_details/templates/gsc_notes_section.html:3]: Not sure I understand.

Initially I assumed that _gsc_notes_ will be like any other includes. Consumer would include this tag at the call site.
in other words, _scenario_assessment_ template will add GSC template tag at the end within its own template.

File: update_supply_chain_information/chain_details/test/test_scd_info.py [hunk 3]
```diff
@@ -41,6 +40,7 @@ def test_ogd_other_dept(self, logged_in_ogd):
         dept_name = "other_dep"
         dept = GovDepartmentFactory(name=dept_name)
         sc = SupplyChainFactory.create(gov_department=dept)
+        ScenarioAssessmentFactory(supply_chain=sc)
```
Comment (dev-digital) [inline update_supply_chain_information/chain_details/test/test_scd_info.py:43]: Just curious, Would the test fail w/o this line ?

File: update_supply_chain_information/supply_chains/admin.py [hunk 4]
```diff
@@ -117,6 +118,44 @@ def gov_department(self, obj):
         return obj.supply_chain.gov_department
 
 
+class ScenarioAssessmentAdmin(admin.ModelAdmin):
+    fields = (
```
Comment (dev-digital) [inline update_supply_chain_information/supply_chains/admin.py:122]: Oh, its going to be long list. Hope it looks fine in admin panel with these many in play.

File: update_supply_chain_information/supply_chains/management/commands/ingestscenarioassessments.py [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+    """Utility to ingest country dependency spreadsheet data"""
+
+    filepath = None
+    unrecognised_supply_chains = set()
+
+    # source_dependency_level_to_choice_value = {
+    #     # Spreadsheet has an empty string for "No dependency" so munge the label in the value lookup table
+    #     choice[1].lower() if choice[1] != "No" else "": choice[0]
+    #     for choice in CountryDependency.DependencyLevel.choices
+    # }
```
Comment (dev-digital) [inline update_supply_chain_information/supply_chains/management/commands/ingestscenarioassessments.py:21]: Remove comments please

File: update_supply_chain_information/supply_chains/migrations/0036_auto_20210921_1240.py [hunk 6]
```diff
@@ -0,0 +1,174 @@
+# Generated by Django 3.2.4 on 2021-09-21 12:40
```
Comment (dev-digital) [inline update_supply_chain_information/supply_chains/migrations/0036_auto_20210921_1240.py:1]: You want to edit the name and dependency as we already have 0036 in the system.

Hang on looks like you have not merged develop into this, have you ??

File: update_supply_chain_information/supply_chains/test/test_ingest_scenario_assessments.py [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+    SupplyChain,
+    ScenarioAssessment,
+    NullableRAGRating,
+)
+
+from supply_chains.test.data import scenario_assessments
+from supply_chains.test.factories import SupplyChainFactory
+
+pytestmark = pytest.mark.django_db
+DATA_FILES_LOC = "update_supply_chain_information/supply_chains/test/data"
```
Comment (dev-digital) [inline update_supply_chain_information/supply_chains/test/test_ingest_scenario_assessments.py:22]: Where is this used ?

**Comment by dev-digital** [comment]
@NickFitz  you might want to add some e2e tests to _scd_info_spec.js_ ?

**Review by dev-digital — COMMENTED** [review]
Added some comments

File: update_supply_chain_information/supply_chains/migrations/0036_auto_20210921_1240.py (hunk 6 above)
Comment (NickFitz) [reply update_supply_chain_information/supply_chains/migrations/0036_auto_20210921_1240.py:1]: Repeatedly. But as it takes too long to rollback the database, delete the out-of-place migrations, make new ones, migrate again, then re-ingest the data, I just added a merge commit  later.

File: update_supply_chain_information/chain_details/templates/chain_details_info.html (hunk 1 above)
Comment (NickFitz) [reply update_supply_chain_information/chain_details/templates/chain_details_info.html:45]: By all means, but given how much is in here already, I'm not increasing the scope of the PR. Getting this stuff into staging at least is fairly urgent at the moment.

File: update_supply_chain_information/chain_details/templates/gsc_notes_section.html (hunk 2 above)
Comment (NickFitz) [reply update_supply_chain_information/chain_details/templates/gsc_notes_section.html:3]: Not sure what you mean? I didn't include the wrapper element in the template tag as that's baking too many assumptions as to its usage in there.

File: update_supply_chain_information/chain_details/test/test_scd_info.py (hunk 3 above)
Comment (NickFitz) [reply update_supply_chain_information/chain_details/test/test_scd_info.py:43]: Yes, as the request uses `ChainDetailsInfoView` and that view's `get_context_data()` will fail unless the SC has a scenario assessment.

File: update_supply_chain_information/supply_chains/test/test_ingest_scenario_assessments.py (hunk 7 above)
Comment (NickFitz) [reply update_supply_chain_information/supply_chains/test/test_ingest_scenario_assessments.py:22]: It isn't; removed.

**Comment by NickFitz** [comment]
I can add E2E tests later as needed, but for the moment we really need to get this into staging.

**Review by dev-digital — APPROVED** [review]

**Pull request merged**